pkg/client/cache/delta_fifo.go

```diff
@@ -404,6 +416,16 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 	defer f.lock.Unlock()
 	for {
 		for len(f.queue) == 0 {
+			// When the queue is empty, invocation of Pop() is blocked until new item is enqueued.
```

Comment:
This means that we see all the queue data before the close actually has any effect? Moreover, as long as we feed the queue with new data, we will never see the close.

We need test cases for this.